From 045de6d1c183afeed7ecb68c44d0e85aa610f580 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Fri, 5 May 2017 14:02:20 +0800
Subject: [PATCH 87/94] dt-bindings: add binding for the Allwinner DWC HDMI
 Encoder

Allwinner SoCs after H3 uses a DesignWare HDMI controller with a
custom PHY.

Add binding document for it.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 .../bindings/display/sunxi/sun8i-h3-dw-hdmi.txt    | 56 ++++++++++++++++++++++
 1 file changed, 56 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/display/sunxi/sun8i-h3-dw-hdmi.txt

diff --git a/Documentation/devicetree/bindings/display/sunxi/sun8i-h3-dw-hdmi.txt b/Documentation/devicetree/bindings/display/sunxi/sun8i-h3-dw-hdmi.txt
new file mode 100644
index 000000000000..8691afc2a88c
--- /dev/null
+++ b/Documentation/devicetree/bindings/display/sunxi/sun8i-h3-dw-hdmi.txt
@@ -0,0 +1,56 @@
+Allwinner DWC HDMI TX Encoder
+============================
+
+The HDMI transmitter is a Synopsys DesignWare HDMI 1.4 TX controller IP
+paired with a custom PHY by Allwiner.
+
+These DT bindings follow the Synopsys DWC HDMI TX bindings defined in
+Documentation/devicetree/bindings/display/bridge/dw_hdmi.txt with the
+following device-specific properties.
+
+Required properties:
+
+- compatible: Shall contain "allwinner,sun8i-h3-dw-hdmi".
+- reg: Should contain two address-size pair. The first is for the DWC HDMI
+  controller and the second is for the custom PHY from Allwinner.
+- reg-io-width: See dw_hdmi.txt. Shall be 1.
+- interrupts: HDMI interrupt number
+- clocks: References to all the clocks specified in the clock-names property
+  as specified in Documentation/devicetree/bindings/clock/clock-bindings.txt.
+- clock-names: Shall contain "iahb" and "isfr" as defined in dw_hdmi.txt and
+    - "iddc" is the clock for HDMI DDC part.
+- ports: See dw_hdmi.txt. The DWC HDMI shall have a single port numbered 0
+  corresponding to the video input of the controller. The port shall have one
+  endpoint connected to the corresponding TCON.
+
+Example:
+
+hdmi: hdmi@1ee0000 {
+	compatible = "allwinner,h3-dw-hdmi";
+	reg = <0x01ee0000 0x10000>,
+	      <0x01ef0000 0x10000>;
+	reg-io-width = <1>;
+	interrupts = <GIC_SPI 88 IRQ_TYPE_LEVEL_HIGH>;
+	clocks = <&ccu CLK_BUS_HDMI>, <&ccu CLK_HDMI>,
+		 <&ccu CLK_HDMI_DDC>;
+	clock-names = "iahb", "isfr", "iddc";
+	resets = <&ccu RST_BUS_HDMI0>, <&ccu RST_BUS_HDMI1>;
+	reset-names = "hdmi", "ddc";
+	status = "disabled";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			hdmi_in: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&tcon0_out_hdmi>;
+			};
+		};
+	};
+};
-- 
2.12.2

